File: ts/256/242.ts
// 哈希表
function isAnagram(s: string, t: string): boolean {
  if (s.length !== t.length) return false
  const ht = {}
  for (const v of s) {
    if (ht[v]) {
      ht[v] += 1
    } else {
      ht[v] = 1
    }
  }
  for (const v of t) {
    if (!ht[v]) return false
    ht[v] -= 1
  }

  for (const k in ht) {
    if (ht[k]) return false
  }
  return true
};

File: py3/256/242.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False
        ht = {}
        for ch in s:
            if ch in ht:
                ht[ch] += 1
            else:
                ht[ch] = 1
        for ch in t:
            if ch in ht:
                ht[ch] -= 1
            else:
                return False
        for ch in ht:
            if ht[ch] != 0:
                return False
        return True